datadog/fwprovider/resource_datadog_integration_gcp_sts_account.go

```diff
@@ -0,0 +1,291 @@
+package fwprovider
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV2"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	frameworkPath "github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ resource.ResourceWithConfigure   = &IntegrationGcpStsAccountResource{}
+	_ resource.ResourceWithImportState = &IntegrationGcpStsAccountResource{}
+)
+
+type IntegrationGcpStsAccountResource struct {
+	Api  *datadogV2.GCPIntegrationApi
+	Auth context.Context
+}
+
+type IntegrationGcpStsAccountModel struct {
+	ID                   types.String `tfsdk:"id"`
+	Automute             types.Bool   `tfsdk:"automute"`
+	ClientEmail          types.String `tfsdk:"client_email"`
+	DelegateAccountEmail types.String `tfsdk:"delegate_account_email"`
+	IsCspmEnabled        types.Bool   `tfsdk:"is_cspm_enabled"`
+	HostFilters          types.Set    `tfsdk:"host_filters"`
+}
+
+func NewIntegrationGcpStsAccountResource() resource.Resource {
+	return &IntegrationGcpStsAccountResource{}
+}
+
+func (r *IntegrationGcpStsAccountResource) Configure(_ context.Context, request resource.ConfigureRequest, response *resource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	r.Api = providerData.DatadogApiInstances.GetGCPIntegrationApiV2()
+	r.Auth = providerData.Auth
+}
+
+func (r *IntegrationGcpStsAccountResource) Metadata(_ context.Context, request resource.MetadataRequest, response *resource.MetadataResponse) {
+	response.TypeName = "integration_gcp_sts_account"
```

Comment:
What do you think about renaming the Terraform resource to `integration_gcp_sts`? I think it would make more sense because ultimately this Terraform resource should [overtake this TF resource](https://registry.terraform.io/providers/DataDog/datadog/latest/docs/resources/integration_gcp), and be the new way to set up the GCP integration going forward.

This TF resource does create a GCP STS account, but it mainly "enables" the integration